### PR TITLE
Update GHA to jdk20

### PR DESCRIPTION
**Pull request by mcimadamore**
This PR updates the github actions on the jdk20 branch

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no review required)

### Reviewers
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - no project role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jextract pull/96/head:pull/96` \
`$ git checkout pull/96`

Update a local copy of the PR: \
`$ git checkout pull/96` \
`$ git pull https://git.openjdk.org/jextract pull/96/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 96`

View PR using the GUI difftool: \
`$ git pr show -t 96`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jextract/pull/96.diff">https://git.openjdk.org/jextract/pull/96.diff</a>

</details>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -38,30 +38,30 @@ jobs:
...
 
-    - name: 'Check Java 19 version'
+    - name: 'Check Java 20 version'
       shell: sh
       run: |
-        ${{ matrix.JAVA19_HOME }}/bin/java --version
+        ${{ matrix.JAVA20_HOME }}/bin/java --version
 
-    - name: 'Setup Java 18'
+    - name: 'Setup Java 19'
```
Comment (mcimadamore) [inline .github/workflows/test.yml:61]: Not sure why this is/was here?

Comment (JornVernee) [reply .github/workflows/test.yml]: We need 2 java versions, one to run Gradle, and one  to build/run jextract. Version 18 was needed to run gradle.

**Review by JornVernee — APPROVED** [review]

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -38,30 +38,30 @@ jobs:
...
-        ${{ matrix.JAVA19_HOME }}/bin/java --version
+        ${{ matrix.JAVA20_HOME }}/bin/java --version
 
-    - name: 'Setup Java 18'
-      uses: oracle-actions/setup-java@v1.1.1
+    - name: 'Setup Java 17'
+      uses: oracle-actions/setup-java@v1.3.1
       with:
-        release: 18
+        release: 17
```
Comment (JornVernee) [inline .github/workflows/test.yml:64]: Maybe we could add a comment here that says that this sets `JAVA_HOME` for gradle to run on?

**Review by JornVernee — APPROVED** [review]

**Pull request closed without merging**